src/assets/scripts/client/aircraft/AircraftModel.js

```diff
@@ -1387,15 +1439,51 @@ export default class AircraftModel {
                 return this._calculateLegalSpeed(this.model.speed.max);
 
             case MCP_MODE.SPEED.VNAV: {
-                const maxSpeed = this.mcp.speed;
-                const waypointSpeed = this.fms.currentWaypoint.speedRestriction;
-                const waypointHasSpeed = waypointSpeed !== -1;
+                // const mcpSpeed = this.mcp.speed;
```

Comment:
nope, just left it there while thinking through the new approach so I could examine the old if I needed to, and forgot to remove. Deleted now.

done